modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -982,6 +983,13 @@ else if (!deleteRelationshipObjectField) {
 
 				throw new RequiredObjectFieldException();
 			}
+
+			if (objectDefinition.isApproved() &&
+				objectDefinition.isModifiable() &&
+				objectDefinition.isSystem()) {
+
+				throw new ApprovedSystemModifiableObjectDefinitionException();
```

Comment:
@mateussgn I'm not sure about the name we should give to this exception,

use UnsupportedOperationException, I saw some examples in objects and makes sense to me for while

Reply:
Ok! I'll change it.